mani_skill/examples/motionplanning/base_motionplanner/motionplanner.py

```diff
@@ -0,0 +1,243 @@
+import mplib
+import numpy as np
+import sapien
+import trimesh
+
+from mani_skill.agents.base_agent import BaseAgent
+from mani_skill.envs.sapien_env import BaseEnv
+from mani_skill.envs.scene import ManiSkillScene
+from mani_skill.utils.structs.pose import to_sapien_pose
+import sapien.physx as physx
+from transforms3d import quaternions
+
+
+class BaseMotionPlanningSolver:
+    OPEN = 1
+    CLOSED = -1
+
+    def __init__(
+        self,
+        env: BaseEnv,
+        debug: bool = False,
+        vis: bool = True,
+        base_pose: sapien.Pose = None,  # TODO mplib doesn't support robot base being anywhere but 0
+        visualize_target_grasp_pose: bool = True,
+        print_env_info: bool = True,
+        joint_vel_limits=0.9,
+        joint_acc_limits=0.9,
+        num_links=7,
```

Comment:
why do we need a num_links argument here? Since other robot mp classes inherit this one anyway, num_links should be a class field like OPEN, CLOSED, call it num_move_group_links or something. I would prefer for it to be hidden as well since user really does not need to modify or read this value (and the naming would matter less as well).